File: Counting_Sort/Counting_Sort.js

// Function For Counting Sort  
function Counting_Sort (a, largest  , arrsize) 
{

            var size = largest +1,
                val;
            var b = Array(size).fill(0);
            for(var i=0 ; i<arrsize;i++)
            {
                 val = a[i];
                b[val] +=1;
            }
            var l = 0, r=0;
            var c =  Array(arrsize).fill(0);

            for (var j = 0; l < size; j++)
            {


                while (b[l] !== 0)
                {
                    c[r] = l;
                    r++;
                    b[l]--;

                }
                l++;
            }
            
            for (var k = 0; k < arrsize; k++)
                
                {
                console.log(c[k]);
                }
}
var a = [1, 7, 3, 4, 2, 6, 8, 5, 9, 11];            //array to be sorted 
//calling sort function 
Counting_Sort(a, 11,10);

File: Counting_Sort/Counting_Sort.dart
import 'dart:io';

class CountingSort {
  List<int> array = [];
  CountingSort(this.array);

  //function to accept values into the array for sorting
  void inputArray(List<int> arr, int size) {
    for (int i = 0; i < size; i++) {
      print('Enter element number $i: ');
      var input = stdin.readLineSync();
      int foo = int.parse(input);
      assert(foo is int && foo > 0); //make sure that the input foo is an integer
      arr.add(foo);
    }
  }


  //function to print the array
  void printArray(List<int> arr) {
    for (var i = 0; i < arr.length; i++)
      print(arr[i]);
  }
  
  
  //function to perform counting sort
  void sortingLogic(List<int> arr, int size) {
    
    //find the value of the largest element
    int max = 0;
    for (int i = 0; i < size; i++) {
      if (arr[i] > max) 
        max = arr[i];
    }
    
    List<int> count = new List(max+1);
    //populate temporary array with zeroes
    for (int i = 0; i < max+1; i++)
      count[i] = 0;
    
    //update newly created count array with number of occurrences 
    for (int i = 0; i < size; i++)
      count[arr[i]]++;
    
    int j = 0;
    for (int i = 0; i < max+1; i++) {
      if (count[i] != 0) {
        int times  = count[i];
        while (times > 0) {
          arr[j++] = i;
          times--;
        }
      }
    }
  }
}

void main() {
  List<int> array = []; //initializing an empty list of integers
  CountingSort countingSort = new CountingSort(array);

  print('How many elements to sort? ');
  var numberOfElements = stdin.readLineSync();
  int size = int.parse(numberOfElements);

  countingSort.inputArray(countingSort.array, size);
  countingSort.sortingLogic(countingSort.array, size);
  
  print('Sorted array is: ');
  countingSort.printArray(countingSort.array);
}

// How many elements to sort?
// 5
// Enter element number 0:
// 23
// Enter element number 1:
// 33
// Enter element number 2:
// 43
// Enter element number 3:
// 53
// Enter element number 4:
// 63
// Sorted array is:
// 23
// 33
// 43
// 53
// 63
